src/ripple/app/ledger/impl/OpenLedger.cpp

```diff
@@ -140,13 +146,29 @@ OpenLedger::apply_one (OpenView& view,
 {
     if (retry)
         flags = flags | tapRETRY;
-    if ((router.getFlags(
-        tx->getTransactionID()) & SF_SIGGOOD) ==
-            SF_SIGGOOD)
-        flags = flags | tapNO_CHECK_SIGN;
+    // If the tx signature has already been checked,
+    // then we'll get a result from the cache. If not,
+    // it'll check now (this can only happen in the case
+    // where we first see the txn during consensus,
+    // OR the cache expires). Exception: If the tx
+    // does not have an sfAccount field, skip the check.
+    // The only transactions that do not need sfAccount,
+    // are also the only transactions that don't get
+    // signed. Any others will be rejected as malformed.
+    if (tx->isFieldPresent(sfAccount) &&
+        router.checkValidity(*tx,
+#if RIPPLE_ENABLE_MULTI_SIGN
```

Comment:
Ugh, the details of multi-signing have infected my nice clean `OpenLedger` class?